src/main/java/com/cdancy/jenkins/rest/JenkinsClient.java

```diff
@@ -52,22 +57,26 @@ public JenkinsClient() {
      */
     public JenkinsClient(@Nullable final String endPoint,
             @Nullable final JenkinsAuthentication authentication,
-            @Nullable final Properties overrides) {
+            @Nullable final Properties overrides,
+            @Nullable final List<Module> modules) {
         this.endPoint = endPoint != null
                 ? endPoint
                 : JenkinsUtils.inferEndpoint();
         this.credentials = authentication != null
                 ? authentication
                 : JenkinsUtils.inferAuthentication();
         this.overrides = mergeOverrides(overrides);
-        this.jenkinsApi = createApi(this.endPoint, this.credentials, this.overrides);
+        this.modules = modules;
+        this.jenkinsApi = createApi(this.endPoint, this.credentials, this.overrides, this.modules);
     }
 
-    private JenkinsApi createApi(final String endPoint, final JenkinsAuthentication authentication, final Properties overrides) {
+    private JenkinsApi createApi(final String endPoint, final JenkinsAuthentication authentication, final Properties overrides, final List<Module> modules) {
+        List<Module> allModules = Lists.newArrayList(new JenkinsAuthenticationModule(authentication));
+        allModules.addAll(modules);
```

Comment:
It's possible this could be null if the developer is instantiating things using the constructor. Lets do a check just to be sure.